ts/WoltLabSuite/WebComponent/woltlab-core-pagination.ts

```diff
@@ -0,0 +1,194 @@
+      div.classList.add(this.#className + "__prev");
+
+      const a = document.createElement("a");
+      a.href = this.getLinkUrl(this.page - 1);
+      a.rel = "prev";
+      a.title = window.WoltLabLanguage.getPhrase("wcf.global.page.previous");
+      a.classList.add("jsTooltip", this.#className + "__link");
+      div.append(a);
+
+      const icon = document.createElement("fa-icon");
+      icon.setIcon("arrow-left");
+      icon.ariaHidden = "true";
+      a.append(icon);
+
+      return div;
+    }
+
+    #getNextLinkElement(): HTMLDivElement | undefined {
+      if (this.page === this.count) {
+        return;
+      }
+
+      const div = document.createElement("div");
+      div.classList.add(this.#className + "__next");
+
+      const a = document.createElement("a");
+      a.href = this.getLinkUrl(this.page + 1);
+      a.rel = "next";
+      a.title = window.WoltLabLanguage.getPhrase("wcf.global.page.next");
+      a.classList.add("jsTooltip", this.#className + "__link");
+      div.append(a);
+
+      const icon = document.createElement("fa-icon");
+      icon.setIcon("arrow-right");
+      icon.ariaHidden = "true";
+      a.append(icon);
+
+      return div;
+    }
+
+    #getLinkItem(page: number): HTMLLIElement {
+      const li = document.createElement("li");
+      li.classList.add(this.#className + "__item");
+
+      const a = document.createElement("a");
+      a.href = this.getLinkUrl(page);
+      a.ariaLabel = `Page ${page}`;
+      a.classList.add(this.#className + "__link");
+      if (page === this.page) {
+        a.ariaCurrent = "page";
+        a.classList.add(this.#className + "__link--current");
+      }
+      a.textContent = page.toString();
+      li.append(a);
+
+      return li;
+    }
+
+    #getLinkItems(): HTMLLIElement[] {
+      const items: HTMLLIElement[] = [];
+
+      let start = this.page - 1;
+      if (start === 3) {
+        start--;
+      }
+      let end = this.page + 1;
+      if (end === this.count - 2) {
+        end++;
+      }
+
+      for (let i = start; i <= end; i++) {
+        if (i <= 1 || i >= this.count) {
+          continue;
+        }
+
+        items.push(this.#getLinkItem(i));
+      }
+
+      return items;
+    }
+
+    #getEllipsesItem(): HTMLLIElement {
+      const li = document.createElement("li");
+      li.classList.add(this.#className + "__item", this.#className + "__item--ellipses");
+
+      const button = document.createElement("button");
+      button.title = window.WoltLabLanguage.getPhrase("wcf.page.jumpTo");
+      button.classList.add("jsTooltip");
+      button.innerHTML = "&ctdot;";
+      button.addEventListener("click", () => {
+        this.dispatchEvent(new Event("jumpToPage"));
+      });
+      li.append(button);
+
+      return li;
+    }
+
+    getLinkUrl(page: number): string {
+      const url = new URL(this.url);
+      url.search += url.search !== "" ? "&" : "?";
+      url.search += new URLSearchParams([["pageNo", page.toString()]]);
```

Comment:
```suggestion
      const url = new URL(this.url);
      url.searchParams.set('pageNo', page.toString());
```

Reply:
This would re-sort the parameters, which is not wanted here.